Snail/Unique.swift

```diff
@@ -22,4 +22,10 @@ public class Unique<T: Equatable>: Variable<T> {
         super.init(value)
         subject.on(.next(value))
     }
+
+    public override func map<U>(transform: @escaping (T) -> U) -> Unique<U> {
+        let newVariable = Unique<U>(transform(value))
+        asObservable().subscribe(onNext: { _ in newVariable.value = transform(self.value) })
+        return newVariable
+    }
```

Comment:
Should we test that this doesn't trigger twice for 2 equivalent values? 
Perhaps something like:

```swift
let firedCount = 0

stringVariable.map { $0.count }.asObservable().subscribe(onNext: { count in
            firedCount += 1
})

stringVariable.value = "HI" // should trigger
stringVariable.value = "HO" // should not trigger

XCTAssertTrue(firedCount == 1)
```

Reply:
done